### PR TITLE
Refactoring for faulty nodes

### LINKED ISSUE #3: Add faulty nodes
Design and implement a way to create some set of faulty nodes within the node pool.

### LINKED ISSUE #13: Spike: How do we identify faulty nodes?
Do we self-identify or do we report values back to coordinator or what? go read that paper.

### PR DESCRIPTION
**Pull request by bethrichardson**
- Coordinator can create/set faulty nodes
- Nodes check for faulty nodes during consensus algorithm
- Coordinator initializes weights to a normalized distribution
- added a base class ConsensusAlgorithm to handle checking for faulty nodes.
- Fixed control messages to be routed correctly
- Handle controlSetFaulty, controlSetAlgorithm messages
- Added tests for faulty nodes and algorithm switching. Updated tests to have a configurable set of nodes.

Closes #3, Closes #13

### CODE REVIEW

File: src/Node.java [hunk 1]
```diff
@@ -62,10 +66,62 @@ public Value validateNetwork(String name){
...
+        else {
+            reply = returnBadAnswer();
+        }
+        lastReply = reply;
+        return reply;
+    }
+
+    public Value[] checkForFaultyNodes(ConsensusAlgorithm algorithm){
+        ConsensusAlgorithm checkNodeAlgorithm;
+        for (int j = 0; j < king.faultySet.length; j++){
```
Comment (bethrichardson) [inline src/Node.java]: change this to reference algorithm.faultySet

File: src/Node.java [hunk 2]
```diff
@@ -62,10 +66,62 @@ public Value validateNetwork(String name){
...
+            checkNodeAlgorithm = new ConsensusAlgorithm(nodeIndex, numNodes, algorithm.faultySet[j], msg, algorithm.w);
+            int alpha = algorithm.calculateAlpha();    //TODO: Need to get a correct alpha calculation here. and add value based on run.
+            checkNodeAlgorithm.run(alpha);
+            algorithm.faultySet[j] = checkNodeAlgorithm.myValue;
+        }
+
+        return algorithm.faultySet;
+    }
+
+    public void updateWeightsKing(){
```
Comment (bethrichardson) [inline src/Node.java]: delete this

**Comment by lifanzhang** [comment]
Reviewed it with @bethrichardson. @fobofindia09, when you get a chance, please take a look at the merge as well in case you catch anything.

**Comment by recreationalcode** [comment]
Mostly looks good to me
There is one thing I'm not sure of .. are we supposed to be checking for a faulty queen?

Looking at the paper:

1) If Pj either does not send a message or sends a message with wrong format in any of the rounds, then Pj is faulty.
2) If myweight > 3/4 in any round and the value sent by the queen in that round is different from myvalue, then the queen is faulty.

**Pull request merged**